pkg/util/deployment/deployment.go

```diff
@@ -189,28 +189,35 @@ func GetAvailablePodsForReplicaSets(c clientset.Interface, rss []*extensions.Rep
 func getReadyPodsCount(pods []api.Pod, minReadySeconds int) int {
 	readyPodCount := 0
 	for _, pod := range pods {
-		if api.IsPodReady(&pod) {
-			// Check if we've passed minReadySeconds since LastTransitionTime
-			// If so, this pod is ready
-			for _, c := range pod.Status.Conditions {
-				// we only care about pod ready conditions
-				if c.Type == api.PodReady {
-					// 2 cases that this ready condition is valid (passed minReadySeconds, i.e. the pod is ready):
-					// 1. minReadySeconds <= 0
-					// 2. LastTransitionTime (is set) + minReadySeconds (>0) < current time
-					minReadySecondsDuration := time.Duration(minReadySeconds) * time.Second
-					if minReadySeconds <= 0 || !c.LastTransitionTime.IsZero() && c.LastTransitionTime.Add(minReadySecondsDuration).Before(time.Now()) {
-						readyPodCount++
-						break
-					}
-				}
-			}
+		if IsPodAvailable(&pod, minReadySeconds) {
+			readyPodCount++
 		}
 	}
 	return readyPodCount
 }
 
-func getPodsForReplicaSets(c clientset.Interface, replicaSets []*extensions.ReplicaSet) ([]api.Pod, error) {
+func IsPodAvailable(pod *api.Pod, minReadySeconds int) bool {
+	if !api.IsPodReady(pod) {
```

Comment:
Since we need to check the Ready condition anyway, does this check add any value?